src/internal/pjsdb/job_crud.go

```diff
@@ -0,0 +1,159 @@
+package pjsdb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+)
+
+const (
+	recursiveTraverseChildren = `
+      WITH RECURSIVE children(parent, id) AS (
+	    -- basecase
+	        SELECT parent, id, 1 as "depth"
+	        FROM pjs.jobs
+	        WHERE id = $1
+	    UNION ALL
+	    -- recursive case
+	        SELECT c.id, j.id, c.depth+1
+	        FROM children c, pjs.jobs j
+	        WHERE j.parent = c.id AND depth < 10000
+	)
+	`
+)
+
+// functions in the CRUD API assume that the JobContext token has already been resolved upstream to a job by the
+// job system. Some functions take a request object such as an IterateJobsRequest. Requests bundle associated fields
+// that can be validated before sql statements are executed.
+
+// CreateJob creates a job entry in postgres.
+func CreateJob(ctx context.Context, tx *pachsql.Tx, req createJobRequest) (JobID, error) {
```

Comment:
It’s not great to have a private type as the argument to a public function. I think that you should just pass the fileset in here.